### PR TITLE
Guardian Fixes/Ranged Simple Animal  Fixes

### PR DESCRIPTION
**Pull request by ghost**
Fixed player controlled ranged simple animals. Their ranged cooldown was handled in automatedmovement, which, following the recent update to skip all that if occupied by a human, was turned off. 

Creating bombs with dense objects will make the bomb dense

The battlecry will only activate on mobs so you can't spam it on stuff with no clickdelay

Slightly nerfed the fire type (takes more damage, spreads less firestacks). Will probably need further balancing. Nerfed healer so it can't run away at top speed dragging someone spamming heal.

Added checks to the healer can't heal while recalled and the bomb type can't create bombs while recalled.

Slightly lowered the damage resistance on the standard version.

The ranged version now fires a very weak projectile (4 damage) at very high speed (click delay). It should be about the same damage output, just without the awkward waiting around for delays/cooldowns which were universally hated on tasers.

Using the communicate verb will now display the message for the user as well so you can keep track of your conversations better.

### CODE REVIEW

**Comment by RemieRichards** [comment]
pls fix ranged

**Comment by KorPhaeron** [comment]
I have no idea how or why that broke. Did you try clicking while recalled or just spam clicks till your cooldown broke?

**Comment by RemieRichards** [comment]
As I told you in Asay, i fired once to show my master my ability, and it simply never worked again.

**Comment by KorPhaeron** [comment]
Sorry was distracted by being one myself at the time. I'll try playing with the ranged values and see what happens.

**Comment by KorPhaeron** [comment]
@RemieRichards 

Figured it out, might have done the fix in an ugly way though, you know how my code is.

**Comment by RemieRichards** [comment]
@KorPhaeron that's an acceptable fix.

File: code/modules/mob/living/simple_animal/guardian/guardian.dm [hunk 1]
```diff
@@ -263,21 +260,26 @@
...
+	projectilesound = 'sound/effects/hit_on_shattered_glass.ogg'
 	ranged = 1
-	rapid = 1
 	range = 13
-	playstyle_string = "As a ranged type, you have only light damage resistance, but are capable of spraying neurotoxin."
+	playstyle_string = "As a ranged type, you have only light damage resistance, but are capable of spraying shards of crystal at incredibly high speed."
 	magic_fluff_string = "..And draw the Sentinel, an alien master of ranged combat."
 	tech_fluff_string = "Boot sequence complete. Ranged combat modules active. Nanoswarm online."
-	bio_fluff_string = "Your scarab swarm finishes mutating and stirs to life, capable of spitting neurotoxin."
+	bio_fluff_string = "Your scarab swarm finishes mutating and stirs to life, capable of spray shards of crystal."
```
Comment (Anonus) [inline code/modules/mob/living/simple_animal/guardian/guardian.dm]: spraying

File: code/modules/mob/living/simple_animal/guardian/guardian.dm [hunk 2]
```diff
@@ -256,6 +254,9 @@
 
 /mob/living/simple_animal/hostile/guardian/healer/AttackingTarget()
 	..()
+	if(src.loc == summoner)
+		src << "<span class='danger'><B>You must be manifested to heal!.</span></B>"
```
Comment (Cheridan) [inline code/modules/mob/living/simple_animal/guardian/guardian.dm]: !.

File: code/modules/mob/living/simple_animal/guardian/guardian.dm [hunk 3]
```diff
@@ -313,6 +319,9 @@
 	var/bomb_cooldown = 0
 
 /mob/living/simple_animal/hostile/guardian/bomb/ShiftClickOn(atom/movable/A)
+	if(src.loc == summoner)
+		src << "<span class='danger'><B>You must be manifested to create bombs!.</span></B>"
```
Comment (Cheridan) [inline code/modules/mob/living/simple_animal/guardian/guardian.dm]: !.

**Pull request merged**